[github-actions] fix existing checks failures due to workflows settings

ZhangLe2016 opened the pull request:
-  Update upload-artifact version from v2 to v4 for handling the build-release check failure:

Error: This request has been automatically failed because it uses a deprecated version of `actions/upload-artifact: v2`.

Learn more: https://github.blog/changelog/2024-02-13-deprecation-notice-v1-and-v2-of-the-artifact-actions/

- Update the ANDROID_NDK_HOME setting for fixing the build apk failure

ANDROID_NDK_HOME not set! Please set it to your Android NDK location!
Error: Process completed with exit code 1.

-  Install coreutils for realpath tool on MacOS for "macos" check failure

script/bootstrap.sh: line 138: realpath: command not found

wgtdkp reviewed (approved):
Thanks! Can you update your PR description and commit message? The changes look like is **not only** about the upload-artifact version

ZhangLe2016 commented:
> Thanks! Can you update your PR description and commit message? The changes look like is **not only** about the upload-artifact version

done, thansk.

ZhangLe2016 commented:
@jwhui I would appreciate it if you could take a look at this PR. It addresses the issues with the test environment that were affecting the checks. Thanks~
